biofefi/pages/feat_importance.py

```diff
@@ -1,192 +1,267 @@
+            ],
+            shap_reduce_data=st.session_state[ConfigStateKeys.ShapDataPercentage],
+            dependent_variable=st.session_state[ConfigStateKeys.DependentVariableName],
+            experiment_name=st.session_state[ConfigStateKeys.ExperimentName],
+            problem_type=st.session_state.get(
+                ConfigStateKeys.ProblemType, ProblemTypes.Auto
+            ).lower(),
+            is_feature_importance=st.session_state[ConfigStateKeys.IsFeatureImportance],
+            # fi_log_dir=
+            angle_rotate_xaxis_labels=st.session_state[
+                PlotOptionKeys.RotateXAxisLabels
+            ],
+            angle_rotate_yaxis_labels=st.session_state[
+                PlotOptionKeys.RotateYAxisLabels
+            ],
+            save_feature_importance_plots=st.session_state[PlotOptionKeys.SavePlots],
+            save_feature_importance_options=st.session_state[
+                ConfigStateKeys.SaveFeatureImportanceOptions
+            ],
+            save_feature_importance_results=st.session_state[
+                ConfigStateKeys.SaveFeatureImportanceResults
+            ],
+            local_importance_methods=st.session_state[
+                ConfigStateKeys.LocalImportanceFeatures
+            ],
+            feature_importance_ensemble=st.session_state[
+                ConfigStateKeys.EnsembleMethods
+            ],
+            global_importance_methods=st.session_state[
+                ConfigStateKeys.GlobalFeatureImportanceMethods
+            ],
+        )
+    fi_opt = fi_opt.parse()
 
-st.session_state[ConfigStateKeys.EnsembleMethods] = ensemble_methods
+    return (
+        fuzzy_opt,
+        fi_opt,
+        st.session_state[ConfigStateKeys.ExperimentName],
+        st.session_state[ExplainModels.ExplainModels],
+    )
 
-st.write("### Local Feature Importance Methods")
-st.write(
-    "Select local methods to interpret individual predictions. "
-    "These methods focus on explaining predictions at a finer granularity."
-)
 
-local_importance_methods = {}
-use_lime = st.checkbox(
-    "LIME",
-    help="Use LIME (Local Interpretable Model-Agnostic Explanations) for local interpretability.",
-)
-local_importance_methods["LIME"] = {"type": "local", "value": use_lime}
-use_local_shap = st.checkbox(
-    "Local SHAP", help="Use SHAP for local feature importance at the instance level."
-)
-local_importance_methods["SHAP"] = {
-    "type": "local",
-    "value": use_local_shap,
-}
-
-st.session_state[ConfigStateKeys.LocalImportanceFeatures] = local_importance_methods
-
-st.write("### Additional Configuration Options")
-
-# Number of important features
-st.number_input(
-    "Number of most important features to plot",
-    min_value=1,
-    value=10,
-    help="Select how many top features to visualise based on their importance score.",
-    key=ConfigStateKeys.NumberOfImportantFeatures,
-)
+def save_upload(file_to_upload: str, content: str, mode: str = "w"):
```

Comment:
This function can be moved to `biofefi.utils.utils` since it's not specific to feature importance.